cdc/sink/flowcontrol/flow_control.go

```diff
@@ -110,6 +119,62 @@ func (c *TableFlowController) Release(resolvedTs uint64) {
 	c.memoryQuota.release(nBytesToRelease)
 }
 
+func (c *TableFlowController) enqueueSingleMsg(
+	msg *model.PolymorphicEvent,
+	commitTs uint64,
+	size uint64,
+) {
+	lastCommitTs := atomic.LoadUint64(&c.lastCommitTs)
+
+	c.queueMu.Lock()
+	defer c.queueMu.Unlock()
+
+	var e deque.Elem
+	// 1. Processing a new transaction.
+	if e = c.queueMu.queue.Back(); e == nil || lastCommitTs < commitTs {
+		c.queueMu.queue.PushBack(&txnSizeEntry{
+			startTs:  msg.StartTs,
+			commitTs: commitTs,
+			size:     size,
+			rowCount: 1,
+		})
+		c.txnsWithSameCommitTs = 1
+		atomic.StoreUint64(&c.lastCommitTs, commitTs)
+		return
+	}
+
+	// Processing txns with the same commitTs.
+	txnEntry := e.(*txnSizeEntry)
+	if txnEntry.commitTs != lastCommitTs {
+		log.Panic("got wrong commitTs from deque, report a bug",
+			zap.Uint64("lastCommitTs", c.lastCommitTs),
+			zap.Uint64("commitTsInDeque", txnEntry.commitTs))
+	}
+
+	// 2. Append row to current transaction entry.
+	if txnEntry.startTs == msg.Row.StartTs &&
```

Comment:
if txnEntry.startTs  != msg.Row.StartTs ,what should be done here ?

Reply:
Then the program will go to the third part and create a new txnEntry, ref: https://github.com/pingcap/tiflow/pull/5281/files#diff-cfe8ccb173f8b93acc1b97cc5825fdb565d539634714fadffacdf18ea4bc8b26R162